### ISSUE TITLE
algo.shortestPaths not supporting INCOMING direction

### ISSUE THREAD

**Issue by sidewayskumara** [neo4j-contrib/neo4j-graph-algorithms#647]
The screenshots below display the directionality of the graph and the result from the following command:

`MATCH (n:Node{name:'Waze'})
CALL algo.shortestPaths.stream(n,null,{direction: "INCOMING"})
YIELD nodeId, distance
WITH nodeId, distance
WHERE algo.isFinite(distance) = true
RETURN nodeId, distance ORDER BY distance DESC, nodeId`

Do I need to do a cypher projection here?

<img width="1572" alt="screenshot 2018-05-24 17 23 38" src="https://user-images.githubusercontent.com/26181282/40473553-708ca736-5f77-11e8-85e6-6fb07affc0da.png">

**Comment by tomasonjo** [comment]
You will need cypher projections here as judging by the code `algo.shortestPaths.stream` does not support direction parameter.

https://github.com/neo4j-contrib/neo4j-graph-algorithms/blob/3.2/algo/src/main/java/org/neo4j/graphalgo/ShortestPathsProc.java#L77

**Comment by sidewayskumara** [comment]
Oh yup I see it now, might be a good idea to make this clearer in the docs as the shortestPaths algorithm docs are clubbed in with the shortestPath algorithm and it seems as if the same parameters apply. Could do a PR if needed.

**Comment by knutwalker** [comment]
I don't think it would be too difficult to support incoming for shortestPaths, though; maybe even for all kinds of shortest path algos. @mneedham WDYT?

**Comment by mneedham** [comment]
@knutwalker sounds like a plan!
